wpilibj/src/sim/java/edu/wpi/first/wpilibj/DriverStation.java

```diff
@@ -145,12 +153,30 @@ public void waitForData() {
    * @param timeout The maximum time in milliseconds to wait.
    */
   public void waitForData(long timeout) {
-    synchronized (m_dataSem) {
+    Lock dataMutex = m_dataMutex;
```

Comment:
Yes, but that only makes sense if the value can change (which it can't if it was declared final).  [Read this](https://en.wikipedia.org/wiki/Cargo_cult_programming).